### ISSUE TITLE
Auto-generate new playlists when full?

### ISSUE THREAD

**Issue by Heavynissen** [brimell/spicetify-star-ratings#12]
Taken from my suggestion from the "Duffey" edition of tihs, I thought I would suggest it here too, as I'm starting to hit the issue now. with around 6-8000 songs rated in most of them.

Is it possible if you max out one of the playlists or more, to make it auto-generate a new one? Maybe called 3.0(2) or something like that? I was just wondering in advance if that happens, what I could do and still keep on rating.

I know the renaming thing is making it a problem, so I'll just let the idea float, and hope there's a solution that works sometime, as this is an extension I can't be without.

**Comment by brimell** [comment]
Yeah tbh that sounds quite easy - just set the max tracks for a playlist to like 5k and then roll over to playlist (1) then playlist (2) etc for the next 5ks

**Comment by brimell** [comment]
Yeah a new playlist could just be automatically created. Also sorry for later replies as I'm just super busy right now so I don't have much time for some of my side projects :(

**Comment by Heavynissen** [comment]
Cool, looking forward to that, and if I hit the mark in the meantime, I'll try the renaming myself and see how it works out. Also no worries at all, you're doing an awesome job keeping this alive, good things take the time it has to take. :D

**Comment by Heavynissen** [comment]
Hey again. Now I have hit the wall at 8000 for 2 of the playlists (the 3.0 and the 4.0) I've tried some testing with renaming and it just won''t work all the way, maybe I'm doing something wrong?

When I hit the 8000 it says "unable to create overflow playlist" if I then move it all to a "3.0 (1)" playlist, I can add to the 3.0 playlist again, but then the ratings are removed for all in the "3.0 (1)" playlist and the other one works as usual again. 
I haven't found a way to keep the ratings, and renaming the original playlist, and creating a new called 3.0 also gets stuck with ""unable to create overflow playlist" 

Any ideas?

**Comment by jullanggit** [comment]
Hey @Heavynissen I have seen some fixes for this in the code, did they work or is this still an issue?

**Comment by Heavynissen** [comment]
Hey @jullanggit I have changed to Tagify some time ago, I can try to to look at it when I get the time, I still use the same folders, so it might be possible

**Comment by Heavynissen** [comment]
I just tried, it looks like it works but still not with the overflow playlists 3.0(1) & 4.0(1)

**Comment by jullanggit** [comment]
@Heavynissen I reworked the overflow playlist system/detection in the [latest release](https://github.com/brimell/spicetify-star-ratings/releases/tag/2.0.2), but don't have enough ratings to actually test it myself, could you check if it now works?

**Comment by Heavynissen** [comment]
@jullanggit Hey, It didn't find the old overflow playlists 3.0(1), or the 4.0(1) playlists. It instead just created a brand new 3.0(1) playlist instead inside the rated folder where the other playlists already are. Also If I try to move songs into that new playlist from the former overflow playlists, it still don't find the stars. Also if I delete that playlist and rate a new song it won't create a new playlist again.

**Comment by jullanggit** [comment]
Hm okay thanks for checking. Are your old playlists also in the ratings folder?

**Comment by Heavynissen** [comment]
Np. Yes they look like this

<img width="437" height="493" alt="Image" src="https://github.com/user-attachments/assets/b465af20-6b15-4815-b8f4-71496c49d084" />

**Comment by jullanggit** [comment]
@Heavynissen Have you restarted Spotify before trying this? We only actively search for playlists on startup and after that assume that they don't change externally (neither additions nor deletions). (This could maybe be improved, but is a reasonable assumption imo, and makes the implementation a bit simpler)

When trying to reproduce this locally, manually creating an overflow playlist, 3.0(1), then restarting did correctly use it afterward.

**Comment by jullanggit** [comment]
Taking a look at it, it wasn't actually that complicated to use up-to-date playlists when adding a rating, so with [2.0.3](https://github.com/brimell/spicetify-star-ratings/releases/tag/2.0.3) the restart should no longer be neccessary. (it works on my machine™)

**Comment by Heavynissen** [comment]
@jullanggit Yeah, I just tried again just now. Right after a restart it does find the overflow playlists, and then it actually moves the songs into the correct overflow playlists. Although it is working, it is just not consistent. It uses up a lot of resources compared to Tagify I guess. I'll try to use them both a couple of days and see how it behaves. The only thing that Tagify doesn't do that this does is moving the songs into the playlists/liked songs, so it would be great if I could get them to work together.

**Comment by jullanggit** [comment]
@Heavynissen with not consistent you mean that it stops working a while after you restart? Or just randomly?

A lot of Resources is also interesting, this extension is for sure not optimized or particularily efficient, but we're also not doing a whole lot, what kind of resources is it using more than tagify?

And for features you might like weighted playback and rating averaging, absolutely essential for me, but not neccessarily for everyone.

In the end this extension cant be everything for everyone, but I do try to make it as nice/good as possible :)

**Comment by Heavynissen** [comment]
@jullanggit It just randomly lags or the stars stops showing for some songs or altogether, then you can click around for a while, and then it's back again. It seems like it's a loading/cache problem if there's a lot of data, but I will say it runs better than star-ratings have ever done, so great work there.

That also ties to the resource "problem". It just feels so heavy for the UI. Tagify changed the storage system from localStorage to IndexedDB, and it got extremely more effective and faster, with no issues. It might be worth looking into?

It's awesome, it's already perfect for me if it was faster and didn't eat so many resources, but that's a small thing. It works, I just think right now it's my extensive library that causes the more "heavy issues"

Hope it makes sense, I'm a bit tired right now :)

**Comment by jullanggit** [comment]
@Heavynissen that makes perfect sense, the current system is not really designed for _a lot_ of ratings, I'll take a look at that when I next get the opportunity, lags like this really shouldn't happen when avoidable in any way. (And there is _a lot_ of potential in terms of optimizations, right now most of the code heavily prefers simplicity and correctness over efficiency, and generally has a sort of "performance shouldn't be that much of an issue" attitude, but hearing that it indeed is, there should be a lot to do about that)

**Comment by jullanggit** [comment]
Just to narrow it down a bit, could you tell me during what kind of usage these issues arise? Is it just when rating a song or some other specific action, or also randomly when you're not really doing anything?

**Comment by Heavynissen** [comment]
@jullanggit Yeah I'll try to come up with some examples.

When rating a song it takes up to around a minute before it has moved the song and the ratings show, whereas Tagify show the rating instantly Just to compare (but Tagify doesn't move anything so that's okay for it to be slower)

Sometimes when I scroll a playlist, random stars disappears, if I then click out of the playlist and go back in they usually are there again. Again it might be some loading issues.

I haven't found any consistency in it though, as right now it is working fine, just a bit slow, and have to wait around a minute before rating something new so it can keep up.

As I can't reproduce the issues right now, I have some screenshots from earlier on, with one playlist having some issues.

It's worse if you rate a song and then click anywhere, but it also just happens while changing playlist, but has been better after Spotify has been running longer.

<img width="675" height="1416" alt="Image" src="https://github.com/user-attachments/assets/1d6c71b5-01f9-4d5f-b427-003ae5811253" />

<img width="730" height="1421" alt="Image" src="https://github.com/user-attachments/assets/5a100f68-bb16-4c54-9a80-042574b8ff2a" />

<img width="682" height="1538" alt="Image" src="https://github.com/user-attachments/assets/71da927f-d473-4a35-b4d4-d07487460859" />

**Comment by jullanggit** [comment]
@Heavynissen I will migrate your performance / scale issues into a new github issue, as they aren't really related to this original issue and it makes my organisation a bit clearer.
